src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,73 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.excepction.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int VALID_AGE = 18;
+    private static final int MIN_LOGIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUserIsNull(user);
+        isAgeNull(user);
+        isAgeValid(user);
+        checkLoginIsNull(user);
+        checkLoginIsValid(user);
+        isPasswordNull(user);
+        checkPasswordIsValid(user);
+        checkIfUserExists(user);
```

Comment:
I'd recommend grouping some of these checks. For example, check all fields on null in one separate method or group login, password and age related checks in separate methods